libyul/optimiser/CallGraphGenerator.cpp

```diff
@@ -29,26 +29,38 @@ using namespace std;
 using namespace dev;
 using namespace yul;
 
+map<YulString, set<YulString>> CallGraphGenerator::callGraph(Block const& _ast)
+{
+	CallGraphGenerator gen;
+	gen(_ast);
+	return gen.m_callGraph;
+}
 
 void CallGraphGenerator::operator()(FunctionalInstruction const& _functionalInstruction)
 {
-	m_callGraph.insert(m_currentFunction, YulString{
-		dev::eth::instructionInfo(_functionalInstruction.instruction).name
-	});
+	string name = dev::eth::instructionInfo(_functionalInstruction.instruction).name;
+	std::transform(name.begin(), name.end(), name.begin(), [](unsigned char _c) { return tolower(_c); });
+	m_callGraph[m_currentFunction].insert(YulString{name});
 	ASTWalker::operator()(_functionalInstruction);
 }
 
 void CallGraphGenerator::operator()(FunctionCall const& _functionCall)
 {
-	m_callGraph.insert(m_currentFunction, _functionCall.functionName.name);
+	m_callGraph[m_currentFunction].insert(_functionCall.functionName.name);
 	ASTWalker::operator()(_functionCall);
 }
 
 void CallGraphGenerator::operator()(FunctionDefinition const& _functionDefinition)
 {
 	YulString previousFunction = m_currentFunction;
 	m_currentFunction = _functionDefinition.name;
+	m_callGraph[m_currentFunction];
```

Comment:
```suggestion
	m_callGraph[m_currentFunction] = {};
```
Here as well - maybe this is easier on the eye?

Reply:
Maybe it should even assert that there was no entry before...

Reply:
Hmpf, how did that change get lost? Why should it assert? This is where the entry is actually created....

Reply:
Ah sorry, now I understood. Yes, will add.